### PR TITLE
Fix for #3835

### PR DESCRIPTION
**Pull request by larseggert**

### CODE REVIEW

**Comment by ianswett** [comment]
I also don't think this is necessary, but I'd like @martinthomson opinion

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -588,37 +588,37 @@ tracks the delivery of data to the application, some of which cannot be observed
...
+          | Recv  | Recv RESET_STREAM
+       +->|       |-----------------------.
+Recv   |  +-------+                       |
+STREAM |    | |                           |
+       +----+ | Recv STREAM + FIN         |
+              v                           |
+          +-------+                       |
+          | Size  | Recv RESET_STREAM     |
+          | Known |---------------------->|
+          +-------+                       |
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md:607]: You receive STREAM frames here as well.

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -588,37 +588,37 @@ tracks the delivery of data to the application, some of which cannot be observed
...
+              |                           |
+              | Recv All Data             |
+              v                           v
+          +-------+ Recv RESET_STREAM +-------+
+          | Data  |--- (optional) --->| Reset |
+          | Recvd |  Recv All Data    | Recvd |
+          +-------+<-- (optional) ----+-------+
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md:614]: And here, in case of reordering or spurious retransmission.

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -493,39 +493,39 @@ Note:
...
+           | Send  | Send RESET_STREAM     |
+       +-->|       |---------------------->|
+Send   |   +-------+                       |
+STREAM |     | |                           |
+       +-----+ | Send STREAM + FIN         |
+               v                           v
+           +-------+                   +-------+
+           | Data  | Send RESET_STREAM | Reset |
+           | Sent  |------------------>| Sent  |
+           +-------+                   +-------+
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md:521]: You send STREAM frames here too when packets get declared lost.

**Review by martinthomson — CHANGES_REQUESTED** [review]
I didn't want to do this.  I believe that more now that I've seen this.  These are both wrong, which demonstrates how this is non-trivial.  That's why this was originally limited to transitions.

**Comment by janaiyengar** [comment]
Right, the problem is that this PR doesn't exhaustively list all loops, but doing so will make this state machine unreadable and useless. The existing FSM shows transitions out of a state, not all the ones that keep the endpoint within that state.

**Comment by larseggert** [comment]
I'm fine with dropping this - I just wanted to move the resolution along by preparing a PR that demonstrated (part of, I guess) the changes needed. @ekr?

**Pull request closed without merging**